### PR TITLE
GeoJson reader handles "id"

### PR DESCRIPTION
**Pull request by SW186000**
## Is this PR related to a proposed Issue?
#344

## What changes were proposed in this PR?
fixed geoJson reader's "id" handling

## How was this patch tested?
add test and test file 

## Did this PR include necessary documentation updates?
no

### CODE REVIEW

File: core/src/main/java/org/datasyslab/geospark/formatMapper/RddReader.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
     {
         SpatialRDD spatialRDD = new SpatialRDD<Geometry>();
         spatialRDD.rawSpatialRDD = rawTextRDD.mapPartitions(formatMapper);
-        spatialRDD.fieldNames = formatMapper.readPropertyNames(rawTextRDD.take(1).get(0).toString());
+        spatialRDD.fieldNames = formatMapper.readColumnNames(rawTextRDD.take(1).get(0).toString());
```
Comment (jiayuasu) [inline core/src/main/java/org/datasyslab/geospark/formatMapper/RddReader.java]: Please don't change the API in RddReader. This PR shouldn't touch this file

File: sql/src/test/resources/bugJson.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"type":"Feature","id":0,"properties":{"zipcode":"00101","name":"A"},"geometry":{"type":"Polygon","coordinates":[[[0.0,0.0],[10.0,0.0],[10.0,10.0],[0.0,10.0],[0.0,0.0]]]}}
```
Comment (jiayuasu) [inline sql/src/test/resources/bugJson.json]: This test data file is not used. Please remove it.

File: core/src/test/java/org/datasyslab/geospark/formatMapper/GeoJsonReaderTest.java [hunk 3]
```diff
@@ -101,4 +101,11 @@ public void testReadToValidGeometryRDD()
         geojsonRDD = GeoJsonReader.readToGeometryRDD(sc, geoJsonWithInvalidGeometries);
         assertEquals(geojsonRDD.rawSpatialRDD.count(), 3);
     }
+
+    @Test
+    public void setGeoJsonIdHandleTest() throws Exception{
+        String pathd = GeoJsonReaderTest.class.getClassLoader().getResource("idContainsGeoJson.json").getPath();
+        SpatialRDD geojsonRDD = GeoJsonReader.readToGeometryRDD(sc,pathd);
+
```
Comment (jiayuasu) [inline core/src/test/java/org/datasyslab/geospark/formatMapper/GeoJsonReaderTest.java]: 1. This test case doesn't actually test the correctness. Please follow the style of other GeoJSON reader test case. For instance, use `assertEquals(geojsonRDD.rawSpatialRDD.count(), 3);` to check the correctness.
2. Please follow the coding style in other test cases. Please put the test data file path in the `onceExecutedBeforeAll()` function

File: sql/src/test/scala/org/datasyslab/geosparksql/adapterTestScala.scala [hunk 4]
```diff
@@ -132,6 +133,12 @@ class adapterTestScala extends TestBaseScala {
       assert (df.columns(1) == "STATEFP")
     }
 
+    it("Read GeoJson that contains id"){
+      val rdd = GeoJsonReader.readToGeometryRDD(sparkSession.sparkContext, "sql/src/test/resources/idContainsGeoJson.json", false, false)
```
Comment (jiayuasu) [inline sql/src/test/scala/org/datasyslab/geosparksql/adapterTestScala.scala]: Please follow the coding style in other test cases and use the proper way to read the input file path

File: sql/src/test/scala/org/datasyslab/geosparksql/adapterTestScala.scala [hunk 5]
```diff
@@ -132,6 +133,12 @@ class adapterTestScala extends TestBaseScala {
...
+      val df = Adapter.toDf(rdd, sparkSession)
+      df.show(false)
```
Comment (jiayuasu) [inline sql/src/test/scala/org/datasyslab/geosparksql/adapterTestScala.scala]: Similarly, this test case doesn't actually test the correctness. Please use `assert` and `count` function to break Spark's lazy evaluation and check the correctness.

File: core/src/main/java/org/datasyslab/geospark/formatMapper/FormatMapper.java [hunk 6]
```diff
@@ -202,10 +202,28 @@ public Geometry readGeoJSON(String geoJson)
...
+    }
+
+
+    public List<String> readColumnNames(String geoString) {
+        List<String> columns = new ArrayList<>();
         switch (splitter){
             case GEOJSON:
-              return readGeoJsonPropertyNames(geoString);
+                if(containsId(geoString)){
+                    columns.add("id");
```
Comment (jiayuasu) [inline core/src/main/java/org/datasyslab/geospark/formatMapper/FormatMapper.java]: The logic here is very simple. Please don't create `containsId` function to check geoString. Just simply check it here.

My concern is, `id` is a pretty common word which may exist in other places of this GeoJSON string. Simply using string contain to check this property may not be safe and can result in another bug.

Comment (SW186000) [reply core/src/main/java/org/datasyslab/geospark/formatMapper/FormatMapper.java]: Thank you for your detailed comments.
I changed the implementation and using GetId method.

Comment (SW186000) [reply core/src/main/java/org/datasyslab/geospark/formatMapper/FormatMapper.java]: Thank you for your comments. 
I try to change my implementation using getId method.

File: core/src/test/resources/idContainsGeoJson.json [hunk 7]
```diff
@@ -0,0 +1 @@
+{"type":"Feature","id":0,"properties":{"zipcode":"00101","name":"A"},"geometry":{"type":"Polygon","coordinates":[[[0.0,0.0],[10.0,0.0],[10.0,10.0],[0.0,10.0],[0.0,0.0]]]}}
```
Comment (jiayuasu) [inline core/src/test/resources/idContainsGeoJson.json]: Is this same as testContainsId.json? Can you please remove all unused json files? Also please make sure,  they all use a consistent name such as "testContainsId".

**Comment by jiayuasu** [comment]
@SW186000 
Hello, thanks for your contribution! The logic of this patch looks good to me now! Please fix the test data file according to my comment. Then I will merge this PR :)

**Pull request merged**